src/Features/Core/Portable/UnusedReferences/ProjectAssets/ProjectAssetsReader.cs

```diff
@@ -19,31 +17,10 @@ internal static class ProjectAssetsReader
         // NuGet will include entries to keep empty folders from being removed. These entries can be ignored.
         private const string NuGetEmptyFileName = "_._";
 
-        public static ImmutableArray<ReferenceInfo> ReadReferences(
-            ImmutableArray<ReferenceInfo> projectReferences,
-            string projectAssetsFilePath)
-        {
-            if (!File.Exists(projectAssetsFilePath))
-            {
-                return ImmutableArray<ReferenceInfo>.Empty;
-            }
-
-            var projectAssetsFileContents = File.ReadAllText(projectAssetsFilePath);
-            ProjectAssetsFile projectAssets;
-
-            try
-            {
-                projectAssets = JsonConvert.DeserializeObject<ProjectAssetsFile>(projectAssetsFileContents);
-            }
-            catch
-            {
-                return ImmutableArray<ReferenceInfo>.Empty;
-            }
-
-            return ReadReferences(projectReferences, projectAssets);
-        }
-
-        internal static ImmutableArray<ReferenceInfo> ReadReferences(
+        /// <summary>
+        /// Enhances references with the assemblies they bring into the compilation and their dependency hierarchy.
+        /// </summary>
+        public static ImmutableArray<ReferenceInfo> EnhanceReferences(
```

Comment:
calling this 'Enhance' is pretty vague, can you make it more like 'AddDependencyHierarchies' or 'AddDependencyInformation' or 'WithDependencyInformation'?  SOmething like that?